src/codegen/compile.ml

```diff
@@ -8209,34 +8209,50 @@ let rec compile_lexp (env : E.t) ae lexp =
   match lexp.it with
   | VarLE var -> Var.set_val env ae var
   | IdxLE (e1, e2) ->
-    let (set_field, get_field) = new_local env "field" in
-    compile_exp_vanilla env ae e1 ^^ (* offset to array *)
-    compile_exp_vanilla env ae e2 ^^ (* idx *)
-    Arr.idx_bigint env ^^
-    set_field ^^
-    get_field,
-    SR.Vanilla,
-    store_ptr ^^
-    (if !Flags.gc_strategy = Mo_config.Flags.Generational
-     then
-      get_field ^^
-      compile_add_const ptr_unskew ^^
-      E.call_import env "rts" "write_barrier"
-     else G.nop)
+      let store_no_barrier = 
```

Comment:
Hmm, this is kinda worse since it generates both version of the code only to throw one away. Want me to have a go?

Reply:
Yes, thanks a lot.